Framework/Crystal/src/FindSXPeaks.cpp

```diff
@@ -84,16 +61,134 @@ void FindSXPeaks::init() {
   declareProperty("EndWorkspaceIndex", EMPTY_INT(), mustBePositive,
                   "End spectrum number  (default FindSXPeaks)");
-  declareProperty("SignalBackground", 10.0,
-                  "Multiplication factor for the signal background");
+
+  // ---------------------------------------------------------------
+  // Peak strategies + Threshold
+  // ---------------------------------------------------------------
+  std::vector<std::string> peakFindingStrategy = {strongestPeakStrategy,
+                                                  allPeaksStrategy};
+  declareProperty(
+      "PeakFindingStrategy", strongestPeakStrategy,
+      boost::make_shared<StringListValidator>(peakFindingStrategy),
+      "Different options for peak finding."
+      "1. StrongestPeakOnly: Looks only for the the strongest peak in each "
+      "spectrum (provided there is "
+      "one). This options is more performant than the AllPeaks option.\n"
+      "2. AllPeaks: This strategy will find all peaks in each "
+      "spectrum. This is slower than StrongestPeakOnly.\n");
+
+  // Declare
+  declareProperty(
+      "SignalBackground", 10.0,
+      "Multiplication factor for the signal background. Peaks which are"
+      " below the estimated background are discarded. The background is "
+      "estimated"
+      " to be an average of the first and the last signal and multiplied"
+      " by the SignalBackground property.\n");
+
+  declareProperty(
+      "AbsoluteBackground", 10.0,
+      "Peaks which are below the specified absolute background are discarded."
+      " The background is gloabally specified for all spectra. Inspect your "
+      "data in the InstrumentView to get a good feeling for the background "
+      "threshold.\n"
+      "Background thresholds which are too low will mistake noise for peaks.");
+
+  // Enable
+  setPropertySettings("SignalBackground",
+                      make_unique<EnabledWhenProperty>(
+                          "PeakFindingStrategy",
+                          Mantid::Kernel::ePropertyCriterion::IS_EQUAL_TO,
+                          strongestPeakStrategy));
+
+  setPropertySettings("AbsoluteBackground",
+                      make_unique<EnabledWhenProperty>(
+                          "PeakFindingStrategy",
+                          Mantid::Kernel::ePropertyCriterion::IS_EQUAL_TO,
+                          allPeaksStrategy));
+
+  // Group
+  const std::string peakGroup = "Peak Finding Settings";
+  setPropertyGroup("PeakFindingStrategy", peakGroup);
+  setPropertyGroup("SignalBackground", peakGroup);
+  setPropertyGroup("AbsoluteBackground", peakGroup);
+
+  // ---------------------------------------------------------------
+  // Resolution
+  // ---------------------------------------------------------------
+  // Declare
+  std::vector<std::string> resolutionStrategy = {
+      relativeResolutionStrategy, absoluteResolutionPeaksStrategy};
+  declareProperty("ResolutionStrategy", relativeResolutionStrategy,
```

Comment:
I tried point 1 on a sample data set. There are two scenarios

### Scenario 1

PeakFindingStrategy: StrongestPeakOnly
SignalBackground: 10

ResolutionStrategy: AbsoluteResolution
TofResolution: 5000
PhiResolution: 1
TowThetaResolution: 1

The resulting peaks workspace (first couple of columns only):

| RunNumber[Y] | DetID[Y] | h[Y] | k[Y] | l[Y] | Wavelength[Y]|...|
|-----------------|-----------|------|-----|-----|----|---|
|0| 1,102,132|0.00|0.00|0.00|2.0346012005009744|...|
|0|1,104,112|0.00|0.00|0.00|2.0030064728788806|...|
|0|1,107,104|0.00|0.00|0.00|1.9822778790598052|...|
|0|1,106,112|0.00|0.00|0.00|1.986706687590025|...|

### Scenario 2

PeakFindingStrategy: StrongestPeakOnly
SignalBackground: 10

ResolutionStrategy: AbsoluteResolution
TofResolution: 1  (<- only thing I changed)
PhiResolution: 1
TowThetaResolution: 1

The resulting peaks workspace (first couple of columns only):

| RunNumber[Y] | DetID[Y] | h[Y] | k[Y] | l[Y] | Wavelength[Y]|...|
|-----------------|-----------|------|-----|-----|----|---|
|0|1,105,104|0.00|0.00|0.00|1.9966571320191759|...|
|0|1,107,104|0.00|0.00|0.00|1.9753950322892595|...|
|0|1,102,132|0.00|0.00|0.00|2.0647578126742214|...|
|0|1,104,112|0.00|0.00|0.00|2.0093560803329833|...|
|0|1,104,124|0.00|0.00|0.00|2.0044459403053558|...|
|0|1,106,108|0.00|0.00|0.00|1.98916083929469|...|
|0|1,106,112|0.00|0.00|0.00|1.986706687590025|...|